src/main/java/com/example/animalshelter/service/AnimalService.java

```diff
@@ -1,8 +1,25 @@
 package com.example.animalshelter.service;
 
+import com.example.animalshelter.controller.AnimalDto;
 import com.example.animalshelter.model.Animal;
 
+/**
+ * Store logic for all operations for an Animal entity.
+ */
 public interface AnimalService {
 
-    boolean add(Animal animal);
+    /**
+     * Persist the Animal object into a database.
+     * @param animal object to be saved
+     * @return same animal if the add action is successful
+     */
+    AnimalDto add(Animal animal);
```

Comment:
Return type should be `Animal`.

Reply:
I was not sure how mapping to AnimalDto object should be resolved. In this situation if our endpoint returns DTO where should we map Animal to AnimalDto? Is this ok to create Mapper class (or in this case just simply pass id) and map result in controller when service return an animal into DTO? Please correct me if I am wrong.